pypsrp/powershell.py

```diff
@@ -1087,11 +1088,18 @@ def invoke(self, input=None, add_to_history=False, apartment_state=None,
         :param remote_stream_options: Whether to return the invocation info on
             the various steams, see complex_objects.RemoteStreamOptions for the
             values. Will default to returning the invocation info on all
+        :param on_error_message: if message is specified, throws WinRMError
+            if any error occurred
         :return: A list of output objects
         """
         self.begin_invoke(input, add_to_history, apartment_state,
                           redirect_shell_error_to_out, remote_stream_options)
-        return self.end_invoke()
+        end_invoke = self.end_invoke()
+        if on_error_message and self.had_errors:
```

Comment:
I extracted this as it would be three times in the code but there is maybe a better place where to put it